SwrveConversationSDK/Conversation/SwrveConversationButton.m

```diff
@@ -1,18 +1,39 @@
 #import "SwrveConversationButton.h"
 #import "SwrveConversationUIButton.h"
-#import "SwrveSetup.h"
 
 @implementation SwrveConversationButton
 
 @synthesize description = _description;
 @synthesize actions = _actions;
 @synthesize target = _target;
 
--(id) initWithTag:(NSString *)tag andDescription:(NSString *)description {
+-(id) initWithTag:(NSString *)tag andDictionary:(NSDictionary *)dict {
     self = [super initWithTag:tag andType:kSwrveControlTypeButton];
     if(self) {
-        _description = description;
+        if([dict objectForKey:kSwrveKeyDescription]) {
+            _description = [dict objectForKey:kSwrveKeyDescription];
+        }
         _target = nil;
+        NSString *target = [dict objectForKey:@"target"]; // Leave the target nil if this a conversation ender (i.e. no following state)
+        if (target && ![target isEqualToString:@""]) {
+            _target = target;
+        }
+        if([dict objectForKey:@"action"]) {
+            _actions = [dict objectForKey:@"action"];
+        }
+
```

Comment:
Make this standard on looking for the keys here as well (target, action)?  And `[string length] != 0` is better than `target && ![target isEqualToString:@""]`, http://stackoverflow.com/a/899214/618320